plugins/spectate.cpp

```diff
@@ -31,25 +33,102 @@ using namespace df::enums;
 
 DFHACK_PLUGIN("spectate");
 DFHACK_PLUGIN_IS_ENABLED(enabled);
+bool dismiss_pause_events = false;
 bool following_dwarf = false;
+df::unit* our_dorf = nullptr;
 void* job_watched = nullptr;
 int32_t timestamp = -1;
 REQUIRE_GLOBAL(world);
 REQUIRE_GLOBAL(ui);
+REQUIRE_GLOBAL(pause_state);
+REQUIRE_GLOBAL(d_init);
+
+// todo: implement as user configurable variables
+#define tick_span 50
```

Comment:
you can steal configuration code from autonestbox

Reply:
steal from autonestbox for the commandline parsing, i mean. you don't need to persist the config for `spectate`

Reply:
So I didn't utilize the lua command line parser, but autonestbox was helpful in setting up a persistent configuration